security-command-center/snippets/src/main/java/management/api/CreateEventThreatDetectionCustomModule.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package management.api;
+
+// [START securitycenter_create_event_threat_detection_custom_module]
+import com.google.cloud.securitycentermanagement.v1.CreateEventThreatDetectionCustomModuleRequest;
+import com.google.cloud.securitycentermanagement.v1.EventThreatDetectionCustomModule;
+import com.google.cloud.securitycentermanagement.v1.EventThreatDetectionCustomModule.EnablementState;
+import com.google.cloud.securitycentermanagement.v1.SecurityCenterManagementClient;
+import com.google.protobuf.ListValue;
+import com.google.protobuf.Struct;
+import com.google.protobuf.Value;
+import java.io.IOException;
+
+public class CreateEventThreatDetectionCustomModule {
+
+  public static void main(String[] args) throws IOException {
+    // https://cloud.google.com/security-command-center/docs/reference/security-center-management/rest/v1/organizations.locations.eventThreatDetectionCustomModules/create
+    // TODO: Developer should replace project_id with a real project ID before running this code
+    String projectId = "project_id";
+
+    String customModuleDisplayName = "custom_module_display_name";
+
+    createEventThreatDetectionCustomModule(projectId, customModuleDisplayName);
+  }
+
+  public static EventThreatDetectionCustomModule createEventThreatDetectionCustomModule(
+      String projectId, String customModuleDisplayName) throws IOException {
+
+    // Initialize client that will be used to send requests. This client only needs
+    // to be created
+    // once, and can be reused for multiple requests.
+    try (SecurityCenterManagementClient client = SecurityCenterManagementClient.create()) {
+
+      String name =
+          String.format(
+              "projects/%s/locations/global/eventThreatDetectionCustomModules/%s",
+              projectId, "custom_module");
```

Comment:
using a constant name for ID is OK for the code sample, but will be a problem for testing. consider providing the module ID as a parameter a use a literal for display name.

Reply:
Actually this sample code is for creating the custom module and we don't have any module ID in place to pass in the parameter, so the module ID will be generated after the creation of custom modules. I have used this constant just as a placeholder. If you want me to change the name then I can use any other name?

Reply:
Addressed, as already mentioned removed this variable name and now using the parent only